nixos/modules/tasks/network-interfaces.nix

```
@@ -1419,6 +1448,8 @@ in
 
     environment.etc.hostid = mkIf (cfg.hostId != null) { source = hostidFile; };
     boot.initrd.systemd.contents."/etc/hostid" = mkIf (cfg.hostId != null) { source = hostidFile; };
+    environment.etc.machine-id = mkIf (cfg.machineId != null) { text = cfg.machineId; };
+    boot.initrd.systemd.contents."/etc/machine-id" = mkIf (cfg.machineId != null) { text = cfg.machineId; };
```

The ID must be newline-terminated
```suggestion
    environment.etc.machine-id = mkIf (cfg.machineId != null) { text = cfg.machineId + "\n"; };
    boot.initrd.systemd.contents."/etc/machine-id" = mkIf (cfg.machineId != null) { text = cfg.machineId + "\n"; };
```